### PR TITLE
[Textual Inversion] Correctly freeze token embeddings

### PR DESCRIPTION
**Pull request by duongna21**
# What does this PR do?

In textual inversion example the token embeddings doesn't seem to be frozen correctly. Zero-ing out the gradient is not sufficient because the effect of weight decay causes non-zero updates. A naive solution would be caching the original token embeddings and re-assign them after every train step. 

## Who can review?
cc @patrickvonplaten @patil-suraj

### CODE REVIEW

File: examples/textual_inversion/textual_inversion.py [hunk 1]
```diff
@@ -392,6 +392,7 @@ def main():
 
     # Initialise the newly added placeholder token with the embeddings of the initializer token
     token_embeds = text_encoder.get_input_embeddings().weight.data
+    original_token_embeds = text_encoder.detach().clone()
```
Comment (destevez-kaleido) [inline examples/textual_inversion/textual_inversion.py]: Hi!
This line should be:
```python
    original_token_embeds = text_encoder.get_input_embeddings().weight.data.detach().clone()
```
or
```python
    original_token_embeds = token_embeds.detach().clone()
```
As you only need a copy of the embeddings matrix

Comment (duongna21) [reply examples/textual_inversion/textual_inversion.py]: @destevez-kaleido Thank you, that's right. Actually I updated this line in a latter commit. Check it out.

Comment (destevez-kaleido) [reply examples/textual_inversion/textual_inversion.py]: You are right, sorry for that. I just realized I had read the commits in the reverse order.

**Comment by patrickvonplaten** [comment]
cc @patil-suraj for textual inversion

**Review by patil-suraj — COMMENTED** [review]
Thanks a lot for the PR!

Did you see any effect on speed with these changes ? Also I'm working on enabling passing embeddings in CLIP which should allow only optimizing the new embeddings and pass them to the model, without having to zero out the gradients. That would be more cleaner solution.

Should have it ready in next few days!

**Comment by duongna21** [comment]
> Did you see any effect on speed with these changes ? 

@patil-suraj Ran both version and there is no difference in terms of speed. 

> Also I'm working on enabling passing embeddings in CLIP which should allow only optimizing the new embeddings and pass them to the model, without having to zero out the gradients. That would be more cleaner solution.

Cool! If it will be released soon, feel free to close this PR.

**Comment by duongna21** [comment]
@patil-suraj I believe this temporary fix should be merged before the new implementation is released.

**Comment by patrickvonplaten** [comment]
@patil-suraj could you take a look here? :-)

**Comment by patrickvonplaten** [comment]
Another gently ping @patil-suraj :-)

**Comment by patrickvonplaten** [comment]
Hey @duongna21, 

Super sorry I didn't see your PR, but now noticed that I merged yesterday pretty much the same logic you implemented here: https://github.com/huggingface/diffusers/pull/1665

Closing this for now. Sorry for not replying here on time :-/

**Comment by duongna21** [comment]
@patrickvonplaten Nevermind. Happy to know that it's fixed.
